src/classes/BGE_DataImportBatchEntry_CTRL.cls

```diff
@@ -208,6 +209,16 @@ public with sharing class BGE_DataImportBatchEntry_CTRL {
                     col.options = BGE_ConfigurationWizard_CTRL.getPicklistOptions(col.fieldName);
 
                     columns.add(col);
+
+                    String targetObject = BDI_DataImportService.getTargetObject(col.fieldName);
+                    if (targetObject == 'Payment') {
```

Comment:
not blocking approval on this thought, but maybe worth moving the mapping of David-object-names to API-names up to a class variable at least? So that it's more visible and more readily managed ...